ISL.ReIdentification.Core/Services/Foundations/OdsDatas/OdsDataService.cs

```diff
@@ -102,12 +102,45 @@ public ValueTask<List<OdsData>> RetrieveAllDecendentsByParentId(Guid odsDataPare
                 .SelectOdsDataByIdAsync(odsDataParentId);
 
             ValidateStorageOdsData(parentRecord, odsDataParentId);
-
             IQueryable<OdsData> query = await this.reIdentificationStorageBroker.SelectAllOdsDatasAsync();
             query = query.Where(ods => ods.OdsHierarchy.IsDescendantOf(parentRecord.OdsHierarchy));
             List<OdsData> descendants = query.ToList();
 
             return descendants;
         });
+
+        public ValueTask<List<OdsData>> RetrieveAllAncestorsByChildId(Guid odsDataChildId) =>
+        TryCatch(async () =>
+        {
+            ValidateOdsDataId(odsDataChildId);
+
+            OdsData childRecord = await this.reIdentificationStorageBroker
+                 .SelectOdsDataByIdAsync(odsDataChildId);
+
+            ValidateStorageOdsData(childRecord, odsDataChildId);
+            OdsData currentNode = childRecord;
+            List<OdsData> ancestors = new List<OdsData>();
```

Comment:
I think the expectation is that it should include the current node i.e. the child and all its ancestors.  We can confirm with DC.

```suggestion
            List<OdsData> ancestors = new List<OdsData>();
            ancestors.Add(currentNode);
```